src/django_components/extensions/debug_highlight.py

```diff
@@ -0,0 +1,131 @@
+
+    # Because the component / slot name is set via styling as a `::before` pseudo-element,
+    # we need to generate a unique ID for each component / slot to avoid conflicts.
+    highlight_id = gen_id()
+
+    output = f"""
+        <style>
+        .{type}-highlight-{highlight_id}::before {{
+            content: "{name}: ";
+            font-weight: bold;
+            color: {color.text_color};
+        }}
+        </style>
+        <div class="{type}-highlight-{highlight_id}" style="border: 1px solid {color.border_color}">
+            {output}
+        </div>
+    """
+
+    return output
+
+
+# TODO - Deprecate `DEBUG_HIGHLIGHT_SLOTS` and `DEBUG_HIGHLIGHT_COMPONENTS` (with removal in v1)
+#        once `extension_defaults` is implemented.
+#        That way people will be able to set the highlighting from single place.
+#        At that point also document this extension in the docs:
+#        - Exposing `ComponentDebugHighlight` from `__init__.py`
+#        - Adding `Component.DebugHighlight` and `Component.debug_highlight` attributes to Component class
+#          so it's easier to find.
+#        - Check docstring of `ComponentDebugHighlight` in the docs and make sure it's correct.
+class HighlightComponentsDescriptor:
+    def __get__(self, obj: Optional[Any], objtype: Type) -> bool:
+        return app_settings.DEBUG_HIGHLIGHT_COMPONENTS
+
+
+class HighlightSlotsDescriptor:
+    def __get__(self, obj: Optional[Any], objtype: Type) -> bool:
+        return app_settings.DEBUG_HIGHLIGHT_SLOTS
+
+
+class ComponentDebugHighlight(ComponentExtension.ExtensionClass):  # type: ignore
+    """
+    The interface for `Component.DebugHighlight`.
+
+    The fields of this class are used to configure the component debug highlighting for this component
+    and its direct slots.
+
+    Read more about [Component debug highlighting](../../concepts/advanced/component_debug_highlighting).
+
+    **Example:**
+
+    ```python
+    from django_components import Component
+
+    class MyComponent(Component):
+        class DebugHighlight:
+            highlight_components = True
+            highlight_slots = True
+    ```
+
+    To highlight ALL components and slots, set
+    [`ComponentsSettings.DEBUG_HIGHLIGHT_SLOTS`](../../settings/components_settings.md#debug_highlight_slots) and
+    [`ComponentsSettings.DEBUG_HIGHLIGHT_COMPONENTS`](../../settings/components_settings.md#debug_highlight_components)
+    to `True`.
+    """
+
+    # TODO_v1 - Remove `DEBUG_HIGHLIGHT_COMPONENTS` and `DEBUG_HIGHLIGHT_SLOTS`
+    #           Instead set this as plain boolean fields.
+    highlight_components = HighlightComponentsDescriptor()
+    """Whether to highlight this component in the rendered output."""
+    highlight_slots = HighlightSlotsDescriptor()
+    """Whether to highlight slots of this component in the rendered output."""
+
+
+# TODO_v1 - Move into standalone extension (own repo?) and ask people to manually add this extension in settings.
```

Comment:
Also, as pointed out here, for v1 it might make sense to move this to it's own repo / project.